project/AgentTest.scala

```diff
@@ -64,24 +75,69 @@ object AgentTest {
               }
             }
             .map { annotation: xsbti.api.Annotation =>
-              val params: Seq[String] = annotation
-                .arguments()
-                .find(_.name() == "parameters")
-                .map(annotationArgument => {
-                  // FIXME
-                  val sanitizedValue =
-                    if (annotationArgument.value().charAt(0) == '"')
-                      annotationArgument.value().substring(1, annotationArgument.value().length - 1)
-                    else
-                      annotationArgument.value()
-                  Seq(sanitizedValue)
-                }) // FIXME
-                .getOrElse(Seq.empty[String])
-              JVMDescription(definition.name(), params)
+              val params: Seq[String] = extractProperty[String](annotation)("parameters").toSeq.flatMap(_.split(" "))
+              val enableJavaAgent: Boolean = extractProperty[Boolean](annotation)("enableJavaAgent").getOrElse(true)
+              JVMDescription(definition.name(), params, enableJavaAgent)
             }
         }
     ).toSeq
   }
 
-  protected case class JVMDescription(className: String, extraParameters: Seq[String])
-}
\ No newline at end of file
+  protected def extractProperty[T: AnnotationPropertyExtractor](annotation: xsbti.api.Annotation)(name: String): Option[T] = {
+    annotation
+      .arguments()
+      .find(_.name() == name)
+      .map(annotationArgument => {
+        implicitly[AnnotationPropertyExtractor[T]].extract(annotationArgument.value())
+      })
+  }
+
+  protected case class JVMDescription(className: String,
+                                      extraParameters: Seq[String],
+                                      enableJavaAgent: Boolean = true) {
+
+    def buildForkOptions(jvmSettings: Seq[String]): ForkOptions = {
+
+      val (jvmSettingsWithAgent: Seq[String], bootJarPaths: Seq[String]) = {
+        if (enableJavaAgent)
+          (jvmSettings, Seq.empty)
+        else
+          settingsWithoutAgent(jvmSettings)
+      }
+
+      val allParameters = jvmSettingsWithAgent ++ extraParameters
+      ForkOptions(runJVMOptions = allParameters, bootJars = bootJarPaths.map(new File(_)))
+    }
+
+    private def settingsWithoutAgent(jvmSettings: Seq[String]): (Seq[String], Seq[String]) = {
+      val (settings, agent) = jvmSettings.partition(param => !(param.startsWith("-javaagent:") && param.contains("io.kamon/agent")))
+      (settings, agent.headOption.map(_.replaceFirst("-javaagent:", "")).toSeq)
+    }
+  }
+
+}
+
+object ExtractPropMonoids {
+
+  trait AnnotationPropertyExtractor[A] {
```

Comment:
@cristianspinetta I think this isn't a monoid, a monoid could be:
```scala

trait Monoid[T] {
  def append(m1:T, m2:T):T
  def idendity:T
}

//String Monoid
val stringMonoid = new Monoid[String] {
  def append(s1:String, s2:String):String = s1 + s2
  val identity = ""
  }
}
```
I think  if we change the name `ExtractPropMonoids` to `ExtractProp` it should be fine. WDYT?